tests/unit/workspace_access/test_scim.py

```diff
@@ -204,6 +204,113 @@ def test_get_apply_task_should_ignore_groups_not_in_migration_state():
     assert sup.get_apply_task(item, MigrationState([mggrp])) is None
 
 
+def test_get_apply_task_target_group_not_found():
```

Comment:
@mohanab-db can you split the additional unit tests into a separate PR? makes it easier to review & revert if necessary

Reply:
@nkvuong Thanks for reviewing. I have removed additional test from this PR and created another PR just with the unit tests - https://github.com/databrickslabs/ucx/pull/1586